rust/main/agents/relayer/src/msg/metadata/base_builder.rs

```diff
@@ -180,63 +181,88 @@ impl BuildsBaseMetadata for BaseMetadataBuilder {
+                                "Could not parse checkpoint syncer config for validator"
+                            );
+                            continue;
+                        };
+
+                        // If this is a LocalStorage based checkpoint syncer and it's not
+                        // allowed, ignore it
+                        if !self.allow_local_checkpoint_syncers
+                            && matches!(config, CheckpointSyncerConf::LocalStorage { .. })
+                        {
+                            debug!(
+                                ?config,
+                                "Ignoring disallowed LocalStorage based checkpoint syncer"
+                            );
+                            continue;
+                        }
+
+                        match config.build_and_validate(None).await {
+                            Ok(checkpoint_syncer) => {
+                                // found the syncer for this validator
+                                return Ok(Some((*validator, checkpoint_syncer)));
+                            }
+                            Err(CheckpointSyncerBuildError::ReorgEvent(reorg_event)) => {
+                                // If a reorg event has been posted to a checkpoint syncer,
+                                // we refuse to build
+                                return Err(CheckpointSyncerBuildError::ReorgEvent(reorg_event));
+                            }
+                            Err(err) => {
+                                debug!(
+                                    error=%err,
+                                    ?config,
+                                    ?validator,
+                                    "Error when loading checkpoint syncer; will attempt to use the next config"
+                                );
+                            }
+                        }
                     }
-                }
-            }
-            if checkpoint_syncers.get(&validator.into()).is_none() {
-                if validator_storage_locations.is_empty() {
-                    warn!(?validator, "Validator has not announced any storage locations; see https://docs.hyperlane.xyz/docs/operators/validators/announcing-your-validator");
-                } else {
                     warn!(
                         ?validator,
                         ?validator_storage_locations,
                         "No valid checkpoint syncer configs for validator"
                     );
-                }
-            }
+                    Ok(None)
+                };
+                Some(future)
+            })
+            .collect::<Vec<_>>();
+
+        let checkpoint_syncers_results = stream::iter(result)
+            .buffer_unordered(10) // Limit the number of concurrent tasks
+            .collect::<Vec<_>>()
+            .await
+            .into_iter()
+            .collect::<Result<Vec<_>, _>>()? // Collect results into a single vector
```

Comment:
This means that if one of the futures gives an Err, that we short circuit / error out of this function right?

This seems like the intended behavior with the whole `Err(CheckpointSyncerBuildError::ReorgEvent(reorg_event));` logic where we want to be conservative and not build a `MultisigCheckpointSyncer` if one of the validators says it's experienced a reorg - but the reason I call it out is that I think it's pretty subtle. Could you maybe add a comment or something about it? Either here or next to the `return Err(CheckpointSyncerBuildError::ReorgEvent(reorg_event));`. I just wanna avoid us introducing more Err opportunities in the future that may not have that intended behavior